File: sop-sdk/sdk-python/request/BaseRequest.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-
from response.BaseResponse import BaseResponse


class BaseRequest:
    """请求类的父类"""

    biz_model = None
    """请求参数"""

    files = None
    """上传文件"""

    def __init__(self):
        pass

    def get_method(self):
        """返回接口名

        :return: 返回接口名
        :rtype: str
        """
        raise Exception('未实现BaseRequest.get_method()方法')

    def get_version(self):
        """返回接口版本号

        :return: 返回版本号，如：1.0
        :rtype: str
        """
        raise Exception('未实现BaseRequest.get_version()方法')

    def get_request_type(self):
        """返回请求类型

        :return: 返回RequestType类实例
        :rtype:  common.RequestType
        """
        raise Exception('未实现BaseRequest.get_request_type()方法')

    def parse_response(self, response_dict):
        response_data = response_dict.get('error_response')
        if response_data is None:
            data_name = self.get_method().replace('.', '_') + '_response'
            response_data = response_dict.get(data_name)
        return response_data

File: sop-sdk/sdk-nodejs/request/BaseRequest.js
const {Class} = require("../common/Class");

/**
 * 请求类父类
 */
exports.BaseRequest = Class.create({
    init: function(){
        this.bizModel = {}
        /*
        [
            {name: 'file1', path: 'd:/dd/1.txt'},
            {name: 'file2', path: 'd:/dd/2.txt'}
        ]
         */
        this.files = []
    },
    /**
     * 返回接口名称
     */
    getMethod: function() {
        throw `未实现BaseRequest类getMethod()方法`;
    },
    /**
     * 返回版本号
     */
    getVersion: function() {
        throw '未实现BaseRequest类getVersion()方法';
    },
    /**
     * 返回请求类型，使用RequestType.js
     */
    getRequestType: function() {
        throw '未实现BaseRequest类getRequestType()方法';
    },
    /**
     * 解析返回结果，子类可以覆盖实现
     * @param responseData 服务器返回内容
     * @returns 返回结果
     */
    parseResponse: function (responseData) {
        let data = responseData['error_response'];
        if (!data) {
            const dataNodeName = this.getMethod().replace(/\./g, '_') + '_response'
            data = responseData[dataNodeName]
        }
        return data;
    }
})
